ml-agents-envs/mlagents/envs/subprocess_env_manager.py

```diff
@@ -78,7 +89,16 @@ def _send_response(cmd_name, payload):
                         texts[brain_name] = action_info.text
                         values[brain_name] = action_info.value
                     all_brain_info = env.step(actions, memories, texts, values)
-                _send_response("step", all_brain_info)
+
+                # The timers in this process are independent from all the processes and the "main" process
+                # So after we send back the root timer, we can safely clear them.
+                # Note that we could randomly return timers a fraction of the time if we wanted to reduce
+                # the data transferred.
+                step_response = StepResponse(
```

Comment:
@harperj  - changed the step to return a NamedTuple with (optional) timing information. This should make it pretty easy to integrate with the async changes (I hope)